unidler.py

```diff
@@ -49,24 +49,32 @@ class RequestHandler(BaseHTTPRequestHandler):
     unidling = {}
 
     def do_GET(self):
-        log.debug(f'HEADERS: {self.headers}')
-
         hostname = self.headers.get('Host', UNIDLER)
         if hostname.startswith(UNIDLER):
-            log.debug('No hostname specified')
+            app_log.debug('No hostname specified')
             self.respond(HTTPStatus.NO_CONTENT, '')
             return
 
+        username = hostname.split('.')[0]
+        log = logging.getLogger('unidler:{}'.format(username))
+
         try:
             if hostname in self.unidling:
+                log.debug('Internal state: unidling is in progress')
                 if self.unidling[hostname].is_done():
                     self.unidling[hostname].enable_ingress()
                     del self.unidling[hostname]
+                else:
+                    log.debug('Unidling is not done yet')
 
             elif is_idle(hostname):
-                self.unidling[hostname] = Unidling(hostname)
+                log.debug('It is idle, so starting unidling')
+                self.unidling[hostname] = Unidling(hostname, log)
                 self.unidling[hostname].start()
 
+            else:
+                log.error('BAD STATE 4')
```

Comment:
I wonder if we could add some more context to this error message so it's more useful without having a look at the code, e.g. `BAD STATE 4: Not unidling and not idled` or something like that.